### PR TITLE
[ONNX] Documentation and typing annotations in registry

### PR DESCRIPTION
**Pull request by justinchuby**
Updating the docstrings and type annotations as I walk through the code.

- Turned some comments into docstrings.
- Added type annotations for some functions in utils and the registry
- Removed direct function imports; importing functions makes name space collision easier to happen and refactoring/code analysis harder: https://google.github.io/styleguide/pyguide.html#22-imports
- Formatted touched files with black

### CODE REVIEW

**Comment by justinchuby** [comment]
cc @thiagocrepaldi, I am not able to add reviewers yet.

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -108,36 +128,66 @@ def setup_onnx_logging(verbose):
         if not is_originally_enabled:
             torch.onnx.disable_log()
 
+
 @contextlib.contextmanager
 def exporter_context(model, mode, verbose):
-    with select_model_mode_for_export(model, mode) as mode_ctx, \
-            disable_apex_o2_state_dict_hook(model) as apex_ctx, \
-            setup_onnx_logging(verbose) as log_ctx:
+    with select_model_mode_for_export(
```
Comment (BowenBao) [inline torch/onnx/utils.py:134]: is this auto-formatted? feels less readable.

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -214,36 +278,48 @@ def _optimize_graph(graph, operator_export_type, _disable_torch_constant_prop=Fa
...
+        graph, params_dict, is_caffe2_aten_fallback()
+    )
     if is_caffe2_aten_fallback():
         # Insert permutes before and after each conv op to ensure correct order.
         torch._C._jit_pass_onnx_quantization_insert_permutes(graph, params_dict)
 
         # Find consecutive permutes that are no-ops and remove them.
-        torch._C._jit_pass_custom_pattern_based_rewrite_graph("""
+        torch._C._jit_pass_custom_pattern_based_rewrite_graph(
+            """
```
Comment (BowenBao) [inline torch/onnx/utils.py:290]: nit: indentation looks a bit weird

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -256,87 +332,121 @@ def _optimize_graph(graph, operator_export_type, _disable_torch_constant_prop=Fa
...
     return graph
 
 
-# We accept dictionaries and strings as ONNX inputs,
-# but they should be only for configuration use.
-# we detect here if these inputs are modified, and if so
-# we warn the user that the changes won't take effect in the
-# traced ONNX graph
 def warn_on_static_input_change(input_states):
+    """Warns the user that the changes won't take effect in the traced ONNX graph.
```
Comment (BowenBao) [inline torch/onnx/utils.py:342]:
```suggestion
    """Warns that changes to input dictionaries and strings won't take effect in the traced ONNX graph.
```

**Comment by BowenBao** [comment]
thanks for the nice clean up, please rebase to resolve conflict.

File: torch/onnx/utils.py (hunk 1 above)
Comment (justinchuby) [reply torch/onnx/utils.py:134]: It is. Black has some discussion on using slashes: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#labels-why-no-backslashes

I agree that this is a case where it feels less readable. But overall, we are trading fine control on how the code looks for freedom from having to worry about formatting in return.

File: torch/onnx/utils.py (hunk 2 above)
Comment (justinchuby) [reply torch/onnx/utils.py:290]: I updated it to

```python
        torch._C._jit_pass_custom_pattern_based_rewrite_graph(
            textwrap.dedent(
                """\
                graph(%Pi):
                    %Pq = quantized::nhwc2nchw(%Pi)
                    %Pr = quantized::nchw2nhwc(%Pq)
                    return (%Pr)"""
            ),
            textwrap.dedent(
                """\
                graph(%Ri):
                    return (%Ri)"""
            ),
            graph,
        )
```

Does it work?

**Comment by justinchuby** [comment]
> thanks for the nice clean up, please rebase to resolve conflict.

Done

**Review by BowenBao — APPROVED** [review]
